components/mediation-monitor/mediation-data-publisher/org.wso2.carbon.prometheus.publisher/src/main/java/org/wso2/carbon/prometheus/publisher/internal/MetricServiceComponent.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.prometheus.publisher.internal;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.service.component.ComponentContext;
+import org.wso2.carbon.base.ServerConfiguration;
+import org.wso2.carbon.prometheus.publisher.service.MetricService;
+import org.wso2.carbon.prometheus.publisher.util.PrometheusPublisherConstants;
+import org.wso2.msf4j.MicroservicesRunner;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @scr.component name="org.wso2.carbon.prometheus.publisher" immediate="true"
+ **/
+
+public class MetricServiceComponent {
+
+    private static final Log LOGGER = LogFactory.getLog(MetricServiceComponent.class);
+
+    private boolean isPublishingEnabled = false;
+    private int port = 1289;
+
+    /**
+     * Activates the component, reads config and starts the metric service
+     *
+     * @param context This is required for the component to be activated as an OSGi component
+     */
+    protected void activate(ComponentContext context) {
+
+        readConfig();
+        if (isPublishingEnabled) {
+            LOGGER.info("Metrics published to Prometheus on port: " + port);
```

Comment:
We are just exposing an endpoint to pull data. aren't we? Shall we change the message to reflect that?